mobile/src/components/course/course.tsx

```diff
@@ -22,19 +24,45 @@ import {
 } from '~/hooks/hooks';
 import { courseModulesActions } from '~/store/actions';
 
+import { Category } from './components/components';
 import { CourseModules } from './components/course-modules/course-modules';
 import { styles } from './styles';
 
 const Course: FC = () => {
   const navigation = useAppNavigate();
   const { width } = useWindowDimensions();
   const dispatch = useAppDispatch();
-  const { course, dataStatus } = useAppSelector((state) => state.courses);
+
+  const { user, course, dataStatus } = useAppSelector(({ auth, courses }) => ({
+    user: auth.user,
+    course: courses.course,
+    dataStatus: courses.dataStatus,
+  }));
+
+  const currentCategory = course?.category;
   const dataCourse = course as CourseGetResponseDto;
```

Comment:
on this screen we have loading so we are not sure that the course will not equal null, we know it only in edit-course-category screen, so to avoid confusing remove this from edit-course-category screen too and add necessary checks where it's needed. Also on edit-course-category screen if there is no course in the store, then we should be navigated back
```suggestion
```

Reply:
Oh, I don't understand anything anymore.
Okay, if we are not sure that the course will not equal null at course.tsx we can add check, like on frontend side:
<img src = "https://user-images.githubusercontent.com/82529236/187458339-084d03a0-6d36-48eb-9a4d-c4d3c66c7364.png" width="60%">
Can I do the same?

At the edit-course-category we know exactly what we have the course.
Why we need delete that   `const courseData = course as CourseGetResponseDto;`
Without that, we will have useless checks...and confusing will be more.